pkg/build/gobuild.go

```diff
@@ -273,15 +284,45 @@ func build(ctx context.Context, ip string, platform v1.Platform, disableOptimiza
 		}
 	}
 
-	cmd.Env = append(defaultEnv, os.Environ()...)
+	args := make([]string, 0, 7)
+	args = append(args, "build")
+	if disableOptimizations {
+		// Disable optimizations (-N) and inlining (-l).
+		args = append(args, "-gcflags", "all=-N -l")
+	}
+	args = addGo113TrimPathFlag(args)
+
+	defaultEnv = append(defaultEnv, os.Environ()...)
+
+	tmpDir, err := ioutil.TempDir("", "ko")
+	if err != nil {
+		return "", err
+	}
+	if os.Getenv("KO_STABLE_OUTPUT") != "" {
+		hasher := md5.New() //nolint: gosec // No strong cryptography needed.
+		hasher.Write([]byte(strings.Join(args, " ") + " " + strings.Join(defaultEnv, " ")))
+
+		tmpDir = filepath.Join(os.TempDir(), "ko", ip, hashInputs(args, defaultEnv))
```

Comment:
should you factor in the working directory?  What if I have multiple `ko` workspaces? 🤔